rlmolecule/alphazero/alphazero_problem.py

```diff
@@ -77,19 +94,70 @@ def reward_wrapper(self, state: GraphSearchState) -> Reward:
                 logger.debug(f"Duplicate reward entry encountered with {state}")
                 self.session.rollback()
 
-        return self.reward_class(reward)
+        return self.reward_class(raw_reward=reward)
+
+    def policy_input_wrapper(self, vertex: AlphaZeroVertex) -> {str: np.ndarray}:
+        """ Cache policy inputs in the vertex class
+
+        :param vertex: A vertex for which policy inputs are desired
+        :return: The policy input dictionary for the captive state
+        """
+        if getattr(vertex, 'policy_inputs', None) is None:
+            vertex.policy_inputs = self.get_policy_inputs(vertex.state)
```

Comment:
@CharlesEdisonTripp, we might want to think about this organization a bit. Right now, the type of Vertex that gets used is up to the Solver (MCTS / AlphaZero), and not the Problem class. There was previously the nice benefit of being able to solve the same Problem class with either MCTS or AlphaZero, but without these strange `getattr` calls, this line would fail since MCTSVertex doesn't expect a `vertex.policy_inputs` attribute. Maybe I could just move these methods to AlphaZero, but it seems odd to have `get_policy_inputs` be a part of `AlphaZeroProblem` while the wrappers are a part of `AlphaZero`?